webui/react/src/hooks/useSettings.ts

```diff
@@ -143,9 +143,16 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
   const { isLoading, querySettings, state, update } = useContext(UserSettings);
   const navigate = useNavigate();
   const pathname = window.location.pathname;
+  const pathnameSaveRoute =
+    config.applicableRoutespace instanceof RegExp
+      ? config.applicableRoutespace.toString()
+      : pathname;
```

Comment:
the `else` condition here should be `config.applicableRoutespace` for consistency with existing behavior: currently `applicableRoutespace` is used for the `storageKey`,  this would instead use `window.location.pathname` as the storage key.

for brevity, we can do:
```suggestion
  const pathnameSaveRoute = String(config.applicableRoutespace ?? '')
```

Reply:
That is ticketed and in progress looks like. https://determinedai.atlassian.net/browse/WEB-752.

Reply:
I don't think the behavior change here was intended, which is why I commented.